webcomponents/tool/src/main/frontend/packages/sakai-rubrics/src/SakaiRubricGradingComment.js

```diff
@@ -101,7 +94,7 @@ export class SakaiRubricGradingComment extends RubricsElement {
         // When we click away from the comment editor we need to save the comment, but only if the comment has been updated
         const updatedComments = commentEditor.getData();
 
-        if (this.criterion.comments !== updatedComments) {
+        if (this.criterion.comments !== updatedComments && !(this.criterion.comments === undefined && updatedComments.trim().length === 0)) {
```

Comment:
this conditional is confusing .... 

1) comments is not equal to updatedComments
2) not(comments is undefined AND updatedComments is empty)

can you simplify?

Reply:
The second part of the conditional (Number 2) is a particular exception to the predominant first. The first is simply looking for a difference. The second excludes (with the NOT) a particular case where the stored comments is not yet defined and the user's contributed comment string is blank. 

Without Number 2, the 'draft' alerts would commonly display for the aforementioned case: where the instructor clicks inside the ckeditor but doesn't type anything and clicks out of the popover.

While Number 2 should be refactored, I don't think it would be clearer. Should we add a comment then?

Reply:
positive logic is easier for most people to understand

const nonEmptyComment = this.criterion.comments !== undefined || updatedComments.trim().length > 0